### PR TITLE
HV-1576 Upgrade the integration tests and the base WildFly version to WildFly 12

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/HV-1576

and then:
 * https://hibernate.atlassian.net/browse/HV-1581
 * https://hibernate.atlassian.net/browse/HV-1582

Better reviewed commit per commit, especially because the first one rejiggers things around.

### CODE REVIEW

**Comment by gsmet** [comment]
Mmmh, there is a failure in the Felix CDI test run on Payara on CI. A bit surprised as I ran the tests locally. They also run successfully on Travis so let's try to trigger the tests again.

**Comment by gsmet** [comment]
Jenkins retest this please

**Comment by gsmet** [comment]
See https://hibernate.atlassian.net/browse/HV-1581 for why the build doesn't fail on Travis or locally. The test fails though, working on fixing it.

**Comment by gsmet** [comment]
Jenkins, retest this please

File: osgi/felixtest/pom.xml [hunk 1]
```diff
@@ -111,6 +111,7 @@
                         <id>felix-integration-test</id>
                         <goals>
                             <goal>integration-test</goal>
+                            <goal>verify</goal>
```
Comment (gunnarmorling) [inline osgi/felixtest/pom.xml]: Uff.

File: engine/pom.xml [hunk 2]
```diff
@@ -215,7 +215,7 @@
                             javax.el.*;version="[2.0.0,4.0.0)";resolution:=optional,
                             org.xml.sax.*;version="0",
                             org.jboss.logging.*;version="[3.1.0,4.0.0)",
-                            com.fasterxml.classmate.*;version="[${classmate.version},2.0.0)",
+                            com.fasterxml.classmate.*;version="[1.3,2.0.0)",
```
Comment (gunnarmorling) [inline engine/pom.xml]: What's the rationale behind that? Fixing it seems like a step back on first sight?

Comment (gsmet) [reply engine/pom.xml]: Well, it's fixed for all the others so I would say it's a step forward for consistency :).

The rationale behing that is that Payara/GlassFish is still using 1.3.3 so upgrading to 1.3.4 makes the Payara/GlassFish CDI/OSGi tests fail.

I could use the build-helper-plugin to extract the major.minor version but I'm not sure it's worth it. And I'm not even sure we would want to require 1.4 if we upgrade to 1.4. That's why I did it like that.

File: osgi/felixtest/pom.xml (hunk 1 above)
Comment (gsmet) [reply osgi/felixtest/pom.xml]: Yeah :). To be fair, it's not the most natural setup.

File: engine/pom.xml (hunk 2 above)
Comment (gsmet) [reply engine/pom.xml]: FWIW, when I will work on making the pom files consistent with JBoss parent/WF parent, the whole range will be parameterized and put in the `<properties>` part of the parent pom; they won't be hidden in engine.

That's how it's done in Search now and I think it's better: we define the range we want but it's still very visible so we don't forget to update it if required.

This is for 6.1 though.

Comment (gunnarmorling) [reply engine/pom.xml]: Ok, that sounds good. On that consistency effort, I hope this won't mean we'll inherit from any JBoss or WF POM, right?

**Comment by gunnarmorling** [comment]
Rebased and applied; thx, @gsmet!

File: engine/pom.xml (hunk 2 above)
Comment (gsmet) [reply engine/pom.xml]: > On that consistency effort, I hope this won't mean we'll inherit from any JBoss or WF POM, right?

Nope. That's the limit we set with Sanne.

**Pull request merged**